### ISSUE TITLE
[Cloud Function] Allow custom subscription name for PubSub trigger type

### ISSUE THREAD

**Issue by ZackTherrien** [hashicorp/terraform-provider-google#11673]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

When creating Cloud Functions in Terraform, you can currently select to trigger your CFs with `PubSub topic publish` event type.
Like so:
```
resource "google_cloudfunctions_function" "my_cf-cloud_function" {
  name        = "my_cf"

  runtime     = "python39"
  available_memory_mb   = 128
  
  source_archive_bucket = google_storage_bucket.bucket.name
  source_archive_object = google_storage_bucket_object.archive.name
  
  event_trigger {
    event_type = "providers/cloud.pubsub/eventTypes/topic.publish"
    resource = "projects/my-project/topics/my-topic"
  }
}
```

However this creates an auto generated subscription name. It would be very beneficial to be able to set the name of the subscription either here or elsewhere after the creation of the event trigger/subscription.

### New or Affected Resource(s)

* google_cloudfunctions_function

### Potential Terraform Configuration

```tf
resource "google_cloudfunctions_function" "my_cf-cloud_function" {
  name        = "my_cf"

  event_trigger {
    event_type = "providers/cloud.pubsub/eventTypes/topic.publish"
    resource = "projects/my-project/topics/my-topic"
    subscription_name = 'SubscriptionForMyCF' # ADD THIS FIELD
  }
}
```

### References

* #0000

**Comment by rileykarson** [comment]
@slevenick in case you want to investigate whether the API supports this

**Comment by slevenick** [comment]
I don't believe this is supported by the API. See https://cloud.google.com/functions/docs/reference/rest/v1/projects.locations.functions#eventtrigger

You can file a request with the product team to add this, but we are unable to do anything on the Terraform side until it exists in the API
